gateways/cybersource/request_builders.go

```diff
@@ -92,6 +114,62 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest) (*Request, error)
 			},
 		},
 	}
+
+	// Apple Pay request
+	if authRequest.Cryptogram != "" {
+		request.PaymentInformation = &PaymentInformation{
+			TokenizedCard: &TokenizedCard{
+				Number: authRequest.CreditCard.Number,
+				ExpirationYear: strconv.Itoa(authRequest.CreditCard.ExpirationYear),
+				ExpirationMonth: fmt.Sprintf("%02d", authRequest.CreditCard.ExpirationMonth),
+				TransactionType: "1",	// in-app transaction
+				Cryptogram: authRequest.Cryptogram,
+			},
+		}
+
+		request.ProcessingInformation.PaymentSolution = "001"	// Apple Pay
+
+		switch authRequest.CreditCard.Network {
+		case sleet.CreditCardNetworkVisa:
+			request.PaymentInformation.TokenizedCard.Type = CardTypeVisa
+			request.ConsumerAuthenticationInformation = &ConsumerAuthenticationInformation{
+				Xid: authRequest.Cryptogram,
+				Cavv: authRequest.Cryptogram,
+			}
+		case sleet.CreditCardNetworkMastercard:
+			request.PaymentInformation.TokenizedCard.Type = CardTypeMastercard
+			request.ProcessingInformation.CommerceIndicator = CommerceIndicatorMastercard
+			request.ConsumerAuthenticationInformation = &ConsumerAuthenticationInformation{
+				UcafAuthenticationData: authRequest.Cryptogram,
+				UcafCollectionIndicator: "2",
+			}
+		case sleet.CreditCardNetworkAmex:
+			request.PaymentInformation.TokenizedCard.Type = CardTypeAmex
+			request.ProcessingInformation.CommerceIndicator = CommerceIndicatorAmex
+			request.ConsumerAuthenticationInformation = &ConsumerAuthenticationInformation{
+				Cavv: authRequest.Cryptogram,
+			}
+
+			// For a 40-byte cryptogram, split the cryptogram into two 20-byte binary values (block A and block B).
+			// Send the first 20-byte value (block A) in the cavv field. Send the second 20-byte value (block B) in
+			// the xid field.
+			if len(authRequest.Cryptogram) == AmexCryptogramMaxLength {
```

Comment:
What happens if the cryptogram is longer than 40 characters? How long is the Apple Pay cryptogram?

Reply:
From https://developer.cybersource.com/api/developer-guides/dita-payments/CreatingOnlineAuth/CreatingAuthReqPNT.html#id189DF0WI0HT_CreatingAuthReqPNT, 
`The cryptogram for the American Express card type is either a 20-byte or a 40-byte binary value. For a 20-byte cryptogram, send the cryptogram in the cavv field.`, it seems Cybersource only handle 20 and 40-byte binary value, so should I fail the request if cryptogram is longer than 40 characters? Or should I assume it not happen?